src/sql.rs

```diff
@@ -711,6 +732,49 @@ pub async fn housekeeping(context: &Context) -> Result<()> {
 
     context.schedule_quota_update().await?;
 
+    let tea_boil_time = Instant::now() + Duration::from_secs(60);
```

Comment:
even though that runs once a day only, on ios, that will virtually always be terminated by the os (as housekeeping will occur most probable when the app is not in foreground). but might also be an issue on some androids.

before doing special things for mobiles here - maybe we should just run much shorter. how many records can be compressed in eg. 5 or 10 seconds? maybe that is still good enough?

Reply:
moreover, maybe also move the loop to the end of housekeeping - in case we are terminated unexpectedly even sooner than eg. 10 seconds, the other, easier, housekeeping is already done.

Reply:
Done. Made it 10s. And also moved setting `Config::LastHousekeeping` to the end of `housekeeping()`, i have no idea why it wasn't there.

Reply:
> And also moved setting Config::LastHousekeeping to the end of housekeeping(), i have no idea why it wasn't there.

hm, thinking it over, it makes sense to  set Config::LastHousekeeping  more at the beginning, i think, this was also the initial gist, then things were added before setting config.

setting Config::LastHousekeeping at the beginning avoids endless loops when things do not work out for whatever reason or are interrupted by the os.

Reply:
Moved to the beginning and added your comment